packages/thirdweb/src/wallets/ecosystem/get-ecosystem-wallet-auth-options.ts

```diff
@@ -0,0 +1,33 @@
+import { getThirdwebBaseUrl } from "../../utils/domains.js";
+import type { AuthOption } from "../types.js";
+import type { EcosystemWalletId } from "../wallet-types.js";
+
+/**
+ * Retrieves the specified auth options for a given ecosystem wallet, if any.
+ * @param walletId The ecosystem wallet ID.
+ * @returns {AuthOption[] | undefined} The auth options for the ecosystem wallet.
+ * @internal
+ */
+export async function getEcosystemWalletAuthOptions(
+  walletId: EcosystemWalletId,
+): Promise<AuthOption[] | undefined> {
+  const headers = new Headers();
+  headers.set("x-ecosystem-id", walletId);
+  const res = await fetch(
+    `${getThirdwebBaseUrl("inAppWallet")}/api/2024-05-05/ecosystem-wallet`,
+    {
+      headers,
+    },
+  );
```

Comment:
you could instead set the header directly in the init options for "headers"

```suggestion
  const res = await fetch(
    `${getThirdwebBaseUrl("inAppWallet")}/api/2024-05-05/ecosystem-wallet`,
    {
      headers: {
        "x-ecosystem-id": walletId
      },
    },
  );
```